### PR TITLE
samples: Bluetooth: direction_finging_connectionless_tx: Add config. and DTS overlay to allow build for nRF52820

### PR DESCRIPTION
**Pull request by ppryga-nordic**
Add configuration and DTS overlay for nRF52830DK_nrf52820 boards.
Required to enable support for nRF52820 SOC.

This PR should be merged after: https://github.com/zephyrproject-rtos/zephyr/pull/36159

### CODE REVIEW

File: samples/bluetooth/direction_finding_connectionless_tx/boards/nrf52833dk_nrf52820.overlay [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+&radio {
+	status = "okay";
+	/* This is an example number of antennas that may be available
+	 * on antenna matrix board.
+	 */
+	dfe-antenna-num = < 10 >;
+	/* This is an example switch pattern that will be used to set an
+	 * antenna for Tx PDU (period before start of Tx CTE).
+	 */
+	dfe-pdu-antenna = <0x1>;
```
Comment (Thalley) [inline samples/bluetooth/direction_finding_connectionless_tx/boards/nrf52833dk_nrf52820.overlay:16]: Shouldn't spacing for values inside of `< >` be consistent?

**Review by Thalley — CHANGES_REQUESTED** [review]

File: samples/bluetooth/direction_finding_connectionless_tx/boards/nrf52833dk_nrf52820.overlay (hunk 1 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_tx/boards/nrf52833dk_nrf52820.overlay]: Yes, it should. It is corrected. Thank you.

File: samples/bluetooth/hci_rpmsg/xx_hci_rpmsg.overlay [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+&radio {
+	status = "okay";
+	/* This is a number of antennas that are available on antenna matrix
+	 * designed by Nordic. For more information see README.rst.
+	 */
+	dfe-antenna-num = < 12 >;
```
Comment (Thalley) [inline samples/bluetooth/hci_rpmsg/xx_hci_rpmsg.overlay:12]:
```suggestion
	dfe-antenna-num = <12>;
```

**Review by Thalley — CHANGES_REQUESTED** [review]

File: samples/bluetooth/hci_rpmsg/xx_hci_rpmsg.overlay (hunk 2 above)
Comment (ppryga) [reply samples/bluetooth/hci_rpmsg/xx_hci_rpmsg.overlay]: The file should not be part of the commit. It is removed.

**Review by Thalley — APPROVED** [review]

**Comment by ppryga** [comment]
@jhedberg @joerchan could you take a look on the PR?

**Review by joerchan — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**